skyportal/tests/frontend/test_tag.py

```diff
@@ -0,0 +1,100 @@
+import uuid
+
+import pytest
+from selenium.webdriver.common.by import By
+
+from skyportal.tests import api
+
+
+def test_add_delete_tag(driver, public_source, super_admin_token, super_admin_user):
+    """Test the basic CRUD operations for object tags."""
+
+    driver.get(f"/become_user/{super_admin_user.id}")
+    tag_name = f"testtag{uuid.uuid4().hex}"
+    status, data = api(
+        "POST",
+        "objtagoption",
+        data={"name": tag_name},
+        token=super_admin_token,
+    )
+    assert status == 200
+
+    tag_id = data["data"]["id"]
+
+    driver.get(f"/source/{public_source.id}")
+    driver.wait_for_xpath(f'//h6[contains(text(), "{public_source.id}")]', timeout=20)
+
+    add_tag_button = driver.wait_for_xpath_to_be_clickable(
+        '//div[contains(@class, "chips")]/following-sibling::button[@data-testid="add-tag-button"] | //div[contains(@class, "chips")]/following-sibling::*//button[@data-testid="add-tag-button"]',
+        timeout=10,
+    )
+    driver.execute_script("arguments[0].click();", add_tag_button)
+    driver.wait_for_xpath('//div[@data-testid="add-tag-dialog"]', timeout=20)
+    driver.wait_for_xpath_to_be_clickable('//div[@data-testid="tag-select"]').click()
+    tag_option = driver.wait_for_xpath(f'//li[@data-testid="tag-option-{tag_id}"]')
+    tag_option.click()
```

Comment:
ah well it's just like this actually, you can do the same thing above instead of the `.execute_script(...)`